influxdb3_client/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+        match status {
+            // TODO - handle the OK response content, return to caller, etc.
+            StatusCode::OK => Ok(()),
+            code => Err(Error::ApiError {
+                code,
+                message: String::from_utf8(content.to_vec())?,
+            }),
+        }
+    }
+}
+
+#[doc(hidden)]
+/// Typestate type for [`WriteRequestBuilder`]
+#[derive(Debug, Copy, Clone)]
+pub struct NoBody;
+
+/// Used to compose a request to the `/api/v3/query_sql` API
+///
+/// Produced by [`Client::api_v3_query_sql`] method.
+#[derive(Debug)]
+pub struct QueryRequestBuilder<'c> {
+    client: &'c Client,
+    db: String,
+    query: String,
+    format: Option<Format>,
+}
+
+// TODO - for now the send method just returns the bytes from the response.
+//   It may be nicer to have the format parameter dictate how we return from
+//   send, e.g., using types more specific to the format selected.
+impl<'c> QueryRequestBuilder<'c> {
+    /// Specify the format, `json`, `csv`, `pretty`, or `parquet`
+    pub fn format(mut self, format: Format) -> Self {
+        self.format = Some(format);
+        self
+    }
+
+    /// Send the request to `/api/v3/query_sql`
+    pub async fn send(self) -> Result<Bytes> {
+        let url = self.client.base_url.join("/api/v3/query_sql")?;
+        let req = QueryParams::from(&self);
+        let mut b = self.client.http_client.get(url).query(&req);
+        if let Some(token) = &self.client.auth_header {
+            b = b.bearer_auth(token.expose_secret());
+        }
+        let resp = b.send().await.map_err(Error::QuerySqlSend)?;
+        let status = resp.status();
+        let content = resp.bytes().await.map_err(Error::Bytes)?;
+
+        match status {
+            StatusCode::OK => Ok(content),
+            code => Err(Error::ApiError {
+                code,
+                message: String::from_utf8(content.to_vec()).map_err(Error::InvalidUtf8)?,
+            }),
+        }
+    }
+}
+
+/// Query parameters for the `/api/v3/query_sql` API
+#[derive(Debug, Serialize)]
+pub struct QueryParams<'a> {
+    db: &'a str,
+    #[serde(rename = "q")]
+    query: &'a str,
+    format: Option<Format>,
+}
+
+impl<'a> From<&'a QueryRequestBuilder<'a>> for QueryParams<'a> {
+    fn from(builder: &'a QueryRequestBuilder<'a>) -> Self {
+        Self {
+            db: &builder.db,
+            query: &builder.query,
+            format: builder.format,
+        }
+    }
+}
+
+/// Output format to request from the server in the `/api/v3/query_sql` API
+#[derive(Debug, Serialize, Copy, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum Format {
+    Json,
+    Csv,
+    Parquet,
+    Pretty,
+}
+
+#[cfg(test)]
+mod tests {
+    use mockito::{Matcher, Server};
+
+    use crate::{Client, Format, Precision};
+
+    #[tokio::test]
+    async fn api_v3_write_lp() {
+        let token = "super-secret-token";
+        let db = "stats";
+        let body = "\
+cpu,host=s1 usage=0.5
```

Comment:
since you're not using a raw string you should be able to indent these lines over no problem since you did the escape on the first line

Reply:
So something like this
```
    let body = "\
        cpu,host=s1 usage=0.5\n\
        cpu,host=s1,region=us-west usage=0.7";
```

Reply:
I will give that a shot - the lack of indentation irks me.